src/pybind/mgr/cephadm/module.py

```diff
@@ -2564,3 +2580,33 @@ def remove_osds_status(self) -> List[OSD]:
         The CLI call to retrieve an osd removal report
         """
         return self.to_remove_osds.all_osds()
+
+    @handle_orch_error
+    def drain_host(self, hostname):
+        # type: (str) -> str
+        """
+        Drain all daemons from a host.
+
+        :param host: host name
+        """
+        self.add_host_label(hostname, '_no_schedule')
+
+        daemons: List[orchestrator.DaemonDescription] = self.cache.get_daemons_by_host(hostname)
+
+        osds_to_remove = []
```

Comment:
```suggestion
        osds_to_remove = [d.daemon_id for d in daemons if d.daemon_type == 'osd ]
```

Possible shorter way of doing it. The way you have it now is fine as well, up to you.